File: archive-application/src/main/java/site/archive/dto/v1/archive/ArchiveListResponseDtoV1.java
package site.archive.dto.v1.archive;

import lombok.Getter;

import java.util.List;

@Getter
public class ArchiveListResponseDtoV1 {

    private final int archiveCount;
    private final List<ArchiveDtoV1> archives;

    private ArchiveListResponseDtoV1(List<ArchiveDtoV1> archives) {
        this.archiveCount = archives.size();
        this.archives = archives;
    }

    public static ArchiveListResponseDtoV1 from(List<ArchiveDtoV1> archiveDtoV1s) {
        return new ArchiveListResponseDtoV1(archiveDtoV1s);
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v1/archive/ArchiveListResponseDtoV1.kt
package site.archive.dto.v1.archive

data class ArchiveListResponseDtoV1(
    val archiveCount: Int,
    val archives: List<ArchiveDtoV1>
) {

    private constructor(archives: List<ArchiveDtoV1>) : this(archives.size, archives)

    companion object {
        @JvmStatic
        fun from(archiveDtoV1s: List<ArchiveDtoV1>): ArchiveListResponseDtoV1 {
            return ArchiveListResponseDtoV1(archiveDtoV1s)
        }
    }

}